kubernetes-itests/src/test/java/io/fabric8/kubernetes/APIServiceIT.java

```diff
@@ -33,6 +37,12 @@ class APIServiceIT {
 
   KubernetesClient client;
 
+  @BeforeEach
+  void ensureExists() {
+    Awaitility.await().atMost(30, TimeUnit.SECONDS)
```

Comment:
Can't this be dealt with `waitUntilCondition`?

Reply:
This statement can also be called `@BeforeClass` since the resource will be loaded and unloaded once per test-suite invocation (as opposed to per test-case)

Reply:
> Can't this be dealt with waitUntilCondition?

Possibly, but I wasn't sure how reliably events were emitted for APIServices since it's a derived resource.

> This statement can also be called @BeforeAll since the resource will be loaded and unloaded once per test-suite invocation (as opposed to per test-case)

We need the client instance - I didn't take the time to look up how to make beforeall non-static.

Reply:
oh, just add a static declaration.

Reply:
anyway, this is fine.

Reply:
> oh, just add a static declaration.

Right, we're already handling that case in the extension.